lib/Sema/TypeCheckStmt.cpp

```diff
@@ -765,10 +765,23 @@ class StmtChecker : public StmtVisitor<StmtChecker, Stmt*> {
 
       // If someone is using an unlabeled break inside of an 'if' or 'do'
       // statement, produce a more specific error.
-      if (S->getTargetName().empty() && !ActiveLabeledStmts.empty() &&
-          (isa<IfStmt>(ActiveLabeledStmts.back()) ||
-           isa<DoStmt>(ActiveLabeledStmts.back())))
-        diagid = diag::unlabeled_break_outside_loop;
+      if (S->getTargetName().empty() && !ActiveLabeledStmts.empty()) {
+        if (isa<IfStmt>(ActiveLabeledStmts.back()) ||
+            isa<DoStmt>(ActiveLabeledStmts.back())) {
+          diagid = diag::unlabeled_break_outside_loop;
+        }
+        
+       
+        if (isa<GuardStmt>(ActiveLabeledStmts.back())) {
+          if (ActiveLabeledStmts.size() >= 2) {
+            auto parent = ActiveLabeledStmts.end()[-2];
+            if (isa<IfStmt>(parent) ||
+                isa<DoStmt>(parent)) {
+              diagid = diag::unlabeled_break_outside_loop;
+            }
+          }
+        }
+      }
```

Comment:
Sorry for nit-picking. This doesn't work for
```swift
func fn(x: Int) {
  if x > 0 {
    gurad x < 1 {
      guard x < 2 {
        break
      }
      return
    }
  }
}
```
The rule here is: If `ActiveLabeledStmts` contains `if` or `do`, we can use `unlabeled_break_outside_loop`.
I think, something like
```cpp
if (S->getTargetName().empty() && 
    std::any_of(ActiveLabeledStmts.rbegin(), ActiveLabeledStmts.rend(),
      [&](Stmt *S) -> bool { return isa<IfStmt>(S) || isa<DoStmt>(S); }))
```
should work.